src/views/virtualmachines/list/components/VMIP/VMIP.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import {
+  V1VirtualMachineInstance,
+  V1VirtualMachineInstanceNetworkInterface,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { Button, ButtonVariant, Popover, PopoverPosition } from '@patternfly/react-core';
+
+type VMIPProps = {
+  vmi: V1VirtualMachineInstance;
+};
+
+const getVMIIPAddresses = (vmi: V1VirtualMachineInstance): string[] => {
+  return [
+    ...new Set(
+      vmi.status?.interfaces
+        ?.filter((i) => !!i.name) // get IPs only for named interfaces because Windows reports IPs for other devices like Loopback Pseudo-Interface 1 etc.
+        .flatMap((i: V1VirtualMachineInstanceNetworkInterface) => {
+          const ips: string[] = [];
+          if (i.ipAddress) {
+            ips.push(i.ipAddress.trim());
+          }
+          if (i.ipAddresses) {
+            ips.push(...i.ipAddresses.map((ip) => ip.trim()));
+          }
+          return ips;
+        })
+        .filter((ip) => ip && ip.length > 0),
+    ),
+  ];
```

Comment:
```suggestion
const namedInterfaces = vmi.status?.interfaces?.filter((i) => !!i.name);
const ipAdressess =  namedInterfaces.flatMap((i) => [i?.ipAddress, ...(i?.ipAdresses || [])]).map((ip) => ip.trim());

const ipAdressesSet = new Set(ipAdresses.filter((ip) => ip && ip != "").map((ip) -> ip.trim));
return [...ipAdressSet];

```
the suggested code does the same thing as the current code.
questions
a. where did you get the documentation about widows IPs ? AFAIK the UI should be "stupid" and show all IPs, we don't need to know about windows quirks,
b. why trim the ips? 
c. according to documentation the `ipAddress` is always the first item in `ipAdresses` , why do you concat the ipAdress to the ipAdresses list ?

Reply:
+1 thanks changed for now

Reply:
+1 for Kobi's suggestion, way more readable for future maintenance, I'd also change the 'i' variable to 'iface' to make it more understandable